apps/src/labs/projects/ProjectManager.ts

```diff
@@ -36,21 +37,21 @@ export default class ProjectManager {
   } = {};
   private lastSource: string | undefined;
   private lastChannel: string | undefined;
+  private timeoutId: number | undefined;
 
   constructor(
-    channelId: string,
     sourcesStore: SourcesStore,
     channelsStore: ChannelsStore,
-    appOptionsStore: AppOptionsStore,
-    getProject: () => Project
+    getProject: () => Project,
+    channelId: string
   ) {
     this.channelId = channelId;
```

Comment:
something I learned recently - you can use `private` (and `readonly`) inside the constructor definition as shorthand for both declaring and assigning these properties, ex:
```
class ProjectManager {
  constructor(private readonly sourcesStore: SourcesStore, private readonly channelsStore, private readonly getProject: () => Project, private readonly channelId: string) {}
...
}
```

Reply:
Oh interesting! Tried it out and it makes you remove them from the list of fields at the top of the class, so it feels a little magical to me and like it would be hard to understand that syntax for someone who didn't know about it. Thoughts on whether we should adopt this pattern? My concern is we want the team to be able to pick up typescript fairly easily.

Reply:
Fair point, yeah - I don't have a strong preference, so fine to keep it this way if it's more readable!